File: Algorithms/Palindrome/Palindrome.py
my_string = input("Enter a string")	#Input a string
if my_string.casefold()==(my_string.casefold())[::-1]:
    print("Palindrome") 

else:
    #Traversing from both ends of the strings and checking if they are same 
    print("NotPalindrome")

File: Algorithms/Palindrome/Palindrome.go
package main

import "fmt"

func isPalindrome(input string) bool {
	for i := 0; i < len(input)/2; i++ {
		if input[i] != input[len(input)-i-1] {
			return false
		}
	}
	return true
}

func main() {
	var str string
	fmt.Println("Insert a string: ")
	fmt.Scanf("%s", &str)
	isPalindrome := isPalindrome(str)

	if isPalindrome {
		fmt.Printf("string %s is palindrome", str)
	} else {
		fmt.Printf("string %s not palindrome", str)
	}
}
